### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by mohamed20163858**
in this subproject I created 3 new classes `Book`, `Rental`, and `Classroom` then I created these associations:-
1- classroom has many students
2-  student belongs to a classroom
3- book has many rentals
4-  rental belongs to a book
5- person has many rentals
6- rental belongs to a person 
and here is the test script used to test the created associations 
```
require './classroom'
require './student'
require './person'
require './book'
require './rental'
moh = Student.new(29, 'Mohamed')
class1 = Classroom.new('1/1')
moh.add_classroom(class1)
puts class1.students[0].name
mos = Student.new(27, 'Mostafa')
class1.add_student(mos)
puts mos.classroom.label
mohsen = Person.new(24, 'Mohsen')
ali = Person.new(30, 'Ali')
book1 = Book.new('journey', 'momo mohsen')
book2 = Book.new('Earth', 'marwa shaheen')
Rental.new('2020-5-4', book1, mohsen)
Rental.new('2021-10-8', book1, ali)
Rental.new('2022-4-17', book2, ali)
puts 'Test persons relationship'
puts 'mohsen:-'
puts mohsen.rentals.count
puts(mohsen.rentals.map { |rental| [rental.date, rental.book.title] })
puts 'Ali:-'
puts ali.rentals.count
puts(ali.rentals.map { |rental| [rental.date, rental.book.title] })
puts 'Test books relationship'
puts 'book1:-'
puts book1.rentals.count
puts(book1.rentals.map { |rental| [rental.date, rental.person.name] })
puts 'book2:-'
puts book2.rentals.count
puts(book2.rentals.map { |rental| [rental.date, rental.person.name] })
``` 
and finally here is a screenshot of the results of running this script:-
image

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (SajjadAhmad14) [inline book.rb:3]: Please add both `Getters` and `Setters` for `@rentals` in the `Book` class.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (SajjadAhmad14) [inline book.rb:4]: Please define a method for adding rentals in `Book` class. You may follow [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) link tp learn more about it.

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
 require './nameable'
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (SajjadAhmad14) [inline person.rb:4]: Please add both `Getters` and `Setters` for `@rentals` in the `Person` class.

File: person.rb [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = nil
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (SajjadAhmad14) [inline person.rb:14]: Please define a method for adding rentals in `Person` class. You may follow [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) link tp learn more about it.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Mohamed, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (mohamed20163858) [reply book.rb:3]: @SajjadAhmad14  there is no need for that if you read this [toturial](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) carefully you will realize all arrays are defined as Getters only

File: book.rb (hunk 2 above)
Comment (mohamed20163858) [reply book.rb:4]: @SajjadAhmad14  no you mustn't there is a many-to-many relationship between the Person class and Book class so to associate them we defined an intermediate class Rental class so all methods to relate Person and Book are inside Rental class if you read my test script and run it you will realize when you call rental class the rentals array inside either Book or Person objects updated automatically

File: person.rb (hunk 3 above)
Comment (mohamed20163858) [reply person.rb:4]: @SajjadAhmad14 agian rentals is an array there is no need to define it as a Setter

File: person.rb (hunk 4 above)
Comment (mohamed20163858) [reply person.rb:14]: @SajjadAhmad14 no you mustn't there is a many-to-many relationship between the Person class and Book class so to associate them we defined an intermediate class Rental class so all methods to relate Person and Book are inside Rental class if you read my test script and run it you will realize when you call rental class the rentals array inside either Book or Person objects updated automatically

**Review by t-yanick — APPROVED** [review]
Hi @mohamed20163858, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**